src/strategies/sonatype.js

```diff
@@ -0,0 +1,93 @@
+// Import Third-Party Dependencies
+import * as httpie from "@myunisoft/httpie";
+
+// Import Internal Dependencies
+import { VULN_MODE } from "../constants.js";
+import { standardizeVulnsPayload } from "./vuln-payload/standardize.js";
+
+// Constants
+const kSonatypeApiURL = "https://ossindex.sonatype.org/api/v3/component-report";
+
+export function SonatypeStrategy() {
+  return {
+    strategy: VULN_MODE.SONATYPE,
+    hydratePayloadDependencies
+  };
+}
+
+/**
+ * @returns {PackageURL} package url, following the Package URL spec semantic
+ * see: https://github.com/package-url/purl-spec
+ */
+function toPackageURL(packageName, packageVersion) {
+  return `pkg:npm/${packageName}@${packageVersion}`;
+}
+
+/**
+ * Coordinates are Sonatype's component identifiers, we must build them
+ * using package's name and different package's versions
+ */
+function createPackageURLCoordinates([dependencyName, dependencyPayload]) {
+  const { versions } = dependencyPayload;
+
+  return versions.map((version) => toPackageURL(dependencyName, version));
+}
+
+async function fetchDataForPackageURLs(coordinates) {
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json; charset=utf-8"
+    },
+    body: JSON.stringify({ coordinates })
+  };
+
+  try {
+    const { data } = await httpie.post(kSonatypeApiURL, requestOptions);
+
+    return JSON.parse(data);
```

Comment:
The response is a string ? If the response http header is `application/json` then httpie should parse it to JSON automatically.

Reply:
Yup you're right. Sonatype returns by default a specific Content-Type `application/vnd.ossindex.component-report.v1+json`. I just added the `"Accept": "application/json"` header to enforce the API to return the `application/json` content-type allowing @myunisoft/httpie to directly parse to JSON.